addons/account/models/account_bank_statement.py

```diff
@@ -152,6 +152,12 @@ def _default_opening_balance(self):
     cashbox_end_id = fields.Many2one('account.bank.statement.cashbox', string="Ending Cashbox")
     is_difference_zero = fields.Boolean(compute='_is_difference_zero', string='Is zero', help="Check if difference is zero.")
 
+    @api.model
+    def create(self, vals):
+        if vals.get('journal_type') == 'bank' and not vals.get('name'):
+            vals['name'] = self.env['ir.sequence'].with_context(ir_sequence_date=vals.get('date') or fields.Date.context_today).next_by_code('account.bank.statement')
```

Comment:
May be this gives an error **invalid input syntax for type date**. You can use: fields.Date.context_today(self) .